File: src/priv/bajdcc/util/lexer/regex/IRegexComponentVisitor.java
package priv.bajdcc.util.lexer.regex;

/**
 * 基于正则表达式组件的访问接口（Visitor模式）
 * 
 * @author bajdcc
 */
public interface IRegexComponentVisitor {

	void visitBegin(Charset node);

	void visitBegin(Constructure node);

	void visitBegin(Repetition node);

	void visitEnd(Charset node);

	void visitEnd(Constructure node);

	void visitEnd(Repetition node);
}

File: src/main/kotlin/com/bajdcc/util/lexer/regex/IRegexComponentVisitor.kt
package com.bajdcc.util.lexer.regex

/**
 * 基于正则表达式组件的访问接口（Visitor模式）
 *
 * @author bajdcc
 */
interface IRegexComponentVisitor {

    fun visitBegin(node: Charset)

    fun visitBegin(node: Constructure)

    fun visitBegin(node: Repetition)

    fun visitEnd(node: Charset)

    fun visitEnd(node: Constructure)

    fun visitEnd(node: Repetition)
}
